check_source.py

```diff
@@ -171,9 +172,15 @@ def check_source_submission(self, source_project, source_package, source_revisio
             )
 
             try:
-                add_role_msg = 'Created automatically from request %s' % self.request.reqid
-                add_role_reqid = create_add_role_request(self.apiurl, source_project, self.required_maintainer,
-                                                         'maintainer', message=add_role_msg)
+                add_roles = get_request_list(self.apiurl, source_project,
+                    req_state=['new', 'review'], req_type='add_role')
+                add_roles = list(filter(self.select_add_role, add_roles))
+                if len(add_roles) > 0:
+                    add_role_reqid = add_roles[0].reqid
+                else:
+                    add_role_msg = 'Created automatically from request %s' % self.request.reqid
+                    add_role_reqid = create_add_role_request(self.apiurl, source_project, self.required_maintainer,
+                                                            'maintainer', message=add_role_msg)
```

Comment:
What about extracting all this to some function like `find_or_create_add_role_req`, `ensure_add_role_req` or whatever name you like? The body of `check_source_submission` is huge already.